internal/worker/worker_test.go

```diff
@@ -247,62 +367,216 @@ func Test_worker_startJobLoop(t *testing.T) {
 		afterFunc  func(args)
 	}
 	defaultCheckFunc := func(w want, got <-chan error) error {
-		if !reflect.DeepEqual(got, w.want) {
-			return errors.Errorf("got = %v, want %v", got, w.want)
+		ecComparator := func(x, y <-chan error) bool {
```

Comment:
x and y are too abstract. I will have to go back to the caller, so rename it appropriately.

Reply:
do you have any idea what name should we use?
I think this func is small enough and should be easy to understand...

Reply:
I recommend that `x is want`, `y is got`

Reply:
actually I followed the comparator implementation that it don't care the order of the parameters.
I will change to not using the comparator implementation

Reply:
fixed